### PR TITLE
Make reactions more intuitive

### PR DESCRIPTION
**Pull request by MinnDevelopment**
[contributing]: https://github.com/DV8FromTheWorld/JDA/wiki/5%29-Contributing

## Pull Request Etiquette

- [x] I have checked the PRs for upcoming features/bug fixes.
- [x] I have read the [contributing guidelines][contributing].

### Changes

- [x] Internal code
- [x] Library interface (affecting end-user code) 
- [ ] Documentation
- [ ] Other: \_____ 

Closes Issue: NaN

## Description

Make usage of reactions a little more intuitive by having some redundancies with better names.

### CODE REVIEW

File: src/main/java/net/dv8tion/jda/internal/utils/EncodingUtil.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        }
+        catch (UnsupportedEncodingException e)
+        {
+            throw new AssertionError(e); // thanks JDK 1.4
+        }
+    }
+
+    public static String encodeCodepointsUTF8(String input)
+    {
+        if (!input.startsWith("U+"))
```
Comment (DV8FromTheWorld) [inline src/main/java/net/dv8tion/jda/internal/utils/EncodingUtil.java]: Can codepoints start with `u+`? or must it be `U+`?

Comment (MinnDevelopment) [reply src/main/java/net/dv8tion/jda/internal/utils/EncodingUtil.java]: I have never seen it being lowercase at least. The common notation is `U+CODE`

Comment (DV8FromTheWorld) [reply src/main/java/net/dv8tion/jda/internal/utils/EncodingUtil.java]: Lets include information about what constitutes a valid format in the message of the `IllegalArgumentException`.
`Invalid Format. Example: "U+1F44D", Provided: "{input}"`

**Comment by MinnDevelopment** [comment]
For some reason I can't add a comment to that review comment, so here is my response instead:
`encodeCodepointsUTF8` is only called by `addReactionById` and only if the code starts with `U+`.

File: src/main/java/net/dv8tion/jda/internal/utils/EncodingUtil.java (hunk 1 above)
Comment (MinnDevelopment) [reply src/main/java/net/dv8tion/jda/internal/utils/EncodingUtil.java]: `encodeCodepointsUTF8` is only called by `addReactionById` and only if the code starts with `U+`.

**Pull request merged**